Creative common licence step not loading

When enabling <step id="cclicense"/> in item-submission.xml configuration, CC licence segment does not appear on submission form.

If a DEBUG level I can see that CC Licence step got prepared : 
`[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfigReader - Submission process config has 2 steps listed.
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfigReader - Loading submission process config named 'Publication'
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfigReader - Submission process config 'Publication' not in cache. Reloading from scratch.
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfig - Added step 'org.dspace.app.rest.submit.step.CollectionStep' as step #0 of submission process Publicat                 ion
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfig - Added step 'org.dspace.app.rest.submit.step.DescribeStep' as step #1 of submission process Publicatio                 n
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfig - Added step 'org.dspace.app.rest.submit.step.DescribeStep' as step #2 of submission process Publicatio                 n
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfig - Added step 'org.dspace.app.rest.submit.step.UploadStep' as step #3 of submission process Publication
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfig - Added step 'org.dspace.app.rest.submit.step.CCLicenseStep' as step #4 of submission process Publicati                 on
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfig - Added step 'org.dspace.app.rest.submit.step.LicenseStep' as step #5 of submission process Publication
[DEBUG] 2021-10-11 19:08:42.833 [https-openssl-apr-8443-exec-3] SubmissionConfigReader - Submission process config has 6 steps listed.
`

Bit later code prints out an Exception related to CC Licence :
`[ERROR] 2021-10-11 19:34:34.756 [https-openssl-apr-8443-exec-9] AInprogressItemConverter - An error occurred during the unmarshal of the data for the section cclicense - reported error: null
java.lang.NullPointerException: null
	at org.dspace.app.rest.converter.ConverterService.toRest(ConverterService.java:107) ~[classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.submit.SubmissionService.getDataCCLicense(SubmissionService.java:307) ~[classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.submit.step.CCLicenseStep.getData(CCLicenseStep.java:41) ~[classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.submit.step.CCLicenseStep.getData(CCLicenseStep.java:26) ~[classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.converter.AInprogressItemConverter.fillFromModel(AInprogressItemConverter.java:105) [classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.converter.WorkspaceItemConverter.convert(WorkspaceItemConverter.java:35) [classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.converter.WorkspaceItemConverter.convert(WorkspaceItemConverter.java:23) [classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.converter.ConverterService.toRest(ConverterService.java:108) [classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.repository.WorkspaceItemRestRepository.findOne(WorkspaceItemRestRepository.java:137) [classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.repository.WorkspaceItemRestRepository.findOne(WorkspaceItemRestRepository.java:75) [classes/:7.1-SNAPSHOT]
	at org.dspace.app.rest.repository.WorkspaceItemRestRepository$$FastClassBySpringCGLIB$$987ae76d.invoke(<generated>) [classes/:7.1-SNAPSHOT]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) [spring-core-5.2.5.RELEASE.jar:5.2.5.RELEASE]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771) [spring-aop-5.2.5.RELEASE.jar:5.2.5.RELEASE]`

In result CC Licence step is not appearing on submission form

Strange is that submission process calls "AInprogressItemConverter" and code expects a CC Licence values to be present in fresh Item object metadata.

--- a/dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/SubmissionService.java
+++ b/dspace-server-webapp/src/main/java/org/dspace/app/rest/submit/SubmissionService.java
@@ -306,7 +306,9 @@ public class SubmissionService {
         result.setRights(creativeCommonsService.getLicenseName(item));
 
         Bitstream licenseRdfBitstream = creativeCommonsService.getLicenseRdfBitstream(item);
-        result.setFile(converter.toRest(licenseRdfBitstream, Projection.DEFAULT));
+        if (licenseRdfBitstream != null) {
+            result.setFile(converter.toRest(licenseRdfBitstream, Projection.DEFAULT));
+        }
 
         return result;
     }

--- a/dspace-server-webapp/src/test/java/org/dspace/app/rest/CCLicenseAddPatchOperationIT.java
+++ b/dspace-server-webapp/src/test/java/org/dspace/app/rest/CCLicenseAddPatchOperationIT.java
@@ -10,7 +10,7 @@ package org.dspace.app.rest;
 import static com.jayway.jsonpath.matchers.JsonPathMatchers.hasJsonPath;
 import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
@@ -118,7 +118,11 @@ public class CCLicenseAddPatchOperationIT extends AbstractControllerIntegrationT
                                               .content(patchBody)
                                               .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
                              .andExpect(status().isOk())
-                             .andExpect(jsonPath("$.sections", not(hasJsonPath("cclicense"))));
+                             .andExpect(jsonPath("$.sections.cclicense", allOf(
+                                     hasJsonPath("$.uri", nullValue()),
+                                     hasJsonPath("$.rights",nullValue()),
+                                     hasJsonPath("$.file", nullValue())
+                             )));
 
 
     }

--- a/dspace-server-webapp/src/test/java/org/dspace/app/rest/CCLicenseRemovePatchOperationIT.java
+++ b/dspace-server-webapp/src/test/java/org/dspace/app/rest/CCLicenseRemovePatchOperationIT.java
@@ -10,7 +10,7 @@ package org.dspace.app.rest;
 import static com.jayway.jsonpath.matchers.JsonPathMatchers.hasJsonPath;
 import static org.hamcrest.Matchers.allOf;
 import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
@@ -95,7 +95,11 @@ public class CCLicenseRemovePatchOperationIT extends AbstractControllerIntegrati
                                               .content(removePatch)
                                               .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
                              .andExpect(status().isOk())
-                             .andExpect(jsonPath("$.sections", not(hasJsonPath("cclicense"))));
+                             .andExpect(jsonPath("$.sections.cclicense", allOf(
+                                     hasJsonPath("$.uri", nullValue()),
+                                     hasJsonPath("$.rights",nullValue()),
+                                     hasJsonPath("$.file", nullValue())
+                             )));
     }
 
 

